packages/beacon-node/src/chain/chain.ts

```diff
@@ -1233,23 +1233,38 @@ export class BeaconChain implements IBeaconChain {
       return;
     }
 
-    // Update custody requirement based on finalized state
-    const stateOrBytes = (await this.getStateOrBytesByCheckpoint(finalizedCheckpoint))?.state;
-
-    if (!stateOrBytes) {
-      throw Error(
-        `No finalized state for epoch ${finalizedCheckpoint.epoch} and root ${finalizedCheckpoint.rootHex} to update target custody group count`
-      );
-    }
-
     // Validators attached to the node
     const validatorIndices = this.beaconProposerCache.getValidatorIndices();
 
+    // Update custody requirement based on finalized state
     let effectiveBalances: number[];
-    if (stateOrBytes instanceof Uint8Array) {
-      effectiveBalances = getEffectiveBalancesFromStateBytes(this.config, stateOrBytes, validatorIndices);
+    const effectiveBalanceIncrements = this.checkpointBalancesCache.get(finalizedCheckpoint);
+    if (effectiveBalanceIncrements) {
+      effectiveBalances = validatorIndices.map(
+        (index) => (effectiveBalanceIncrements[index] ?? 0) * EFFECTIVE_BALANCE_INCREMENT
+      );
     } else {
-      effectiveBalances = validatorIndices.map((index) => stateOrBytes.validators.get(index).effectiveBalance);
+      // If there's no cached effective balances, get the state from disk and parse them out
+      this.logger.debug("No cached finalized effective balances to update target custody group count", {
+        finalizedEpoch: finalizedCheckpoint.epoch,
+        finalizedRoot: finalizedCheckpoint.rootHex,
+      });
+
+      const stateOrBytes = (await this.getStateOrBytesByCheckpoint(finalizedCheckpoint))?.state;
+      if (!stateOrBytes) {
+        // If even the state is not available, we cannot update the custody group count
+        this.logger.debug("No finalized state or bytes available to update target custody group count", {
```

Comment:
I think this should never happen, maybe @twoeths can confirm, there might be some edge case. In any case logging to debug is ok I'd assume as updating cgc isn't really time sensitive

Reply:
it should never happen, but for type-safety its there